### PR TITLE
Fix public link share, default expiration behavior

### PR DESCRIPTION
**Pull request by karakayasemi**
## Description
- "default link expiration" sub-entry location problem fixed. Now, it is under "Set expiration date".

- Currently, when the expiration date enabled and not enforced,  the default expiration date is set even though the user deleted it on share creation. With this PR, if the user removes the expiration date on share creation, no expiration date will be set as discussed in here: https://github.com/owncloud/enterprise/issues/3204#issuecomment-478657949

## Related Issue
https://github.com/owncloud/enterprise/issues/3204

## Motivation and Context
Fixing bugs.

## How Has This Been Tested?
Unit tests are adjusted. Also, manually tested with following steps:
- Activate, "Set expiration date" from admin sharing settings. The expiration date sub-entry should be listed under "Set expiration date" input.

To test share creation behavior;
- Activate, "Set expiration date" but not enforce.
- Click share a file via public link and remove default expiration date from its input field. The share should be created without an expiration date.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: settings/templates/panels/admin/filesharing.php [hunk 1]
```diff
@@ -56,15 +56,28 @@
...
 
 		<input type="checkbox" name="shareapi_allow_public_notification" id="allowPublicMailNotification" class="checkbox"
 			   value="1" <?php if ($_['allowPublicMailNotification'] == 'yes') {
 	print_unescaped('checked="checked"');
 } ?> />
 		<label for="allowPublicMailNotification"><?php p($l->t('Allow users to send mail notification for shared files'));?></label><br/>
-	<span id="publicMailNotificationLang" <?php if ($_['allowPublicMailNotification'] == 'no') {
-	print_unescaped('class="hidden"');
-} ?>>
+	<span id="publicMailNotificationLang" class="indent <?php if ($_['allowPublicMailNotification'] == 'no') {
```
Comment (karakayasemi) [inline settings/templates/panels/admin/filesharing.php]: add missing indent

**Comment by karakayasemi** [comment]
@ownclouders rebase

**Comment by karakayasemi** [comment]
@ownclouders rebase

**Comment by karakayasemi** [comment]
@ownclouders rebase

**Comment by karakayasemi** [comment]
@ownclouders rebase

File: settings/templates/panels/admin/filesharing.php [hunk 2]
```diff
@@ -56,19 +56,28 @@
...
 		<label for="shareapiDefaultExpireDate"><?php p($l->t('Set default expiration date'));?></label><br/>
+		<span id="setDefaultExpireDate" class="indent <?php if ($_['allowLinks'] !== 'yes' || $_['shareDefaultExpireDateSet'] === 'no' || $_['shareAPIEnabled'] === 'no') {
+	p('hidden');
+}?>">
+			<?php p($l->t('Expire after ')); ?>
+			<input type="text" name='shareapi_expire_after_n_days' id="shareapiExpireAfterNDays" placeholder="<?php p('7')?>"
+				   value='<?php p($_['shareExpireAfterNDays']) ?>' />
+			<?php p($l->t('days')); ?>
+			<input type="checkbox" name="shareapi_enforce_expire_date" id="shareapiEnforceExpireDate" class="checkbox"
+				   value="1" <?php if ($_['shareEnforceExpireDate'] === 'yes'): ?>checked="checked"<?php endif; ?>/>
```
Comment (jvillafanez) [inline settings/templates/panels/admin/filesharing.php]: Needs better readability here.
```
<input type="checkbox" name="shareapi_enforce_expire_date" id="shareapiEnforceExpireDate" class="checkbox"
					   value="1" <?php if ($_['shareEnforceExpireDate'] === 'yes'){ p('checked="checked"'); }?> />
```
or
```
<?php if ($_['shareEnforceExpireDate'] === 'yes'): ?>
  <input type="checkbox" name="shareapi_enforce_expire_date" id="shareapiEnforceExpireDate" class="checkbox" value="1" checked="checked" />
<?php else: ?>
  <input type="checkbox" name="shareapi_enforce_expire_date" id="shareapiEnforceExpireDate" class="checkbox" value="1" />
<?php endif; ?>
```
Taking into account that the code checker will likely complain about the oneliner (and force you to split it into several lines), I'm more inclined to use the second option.

Comment (karakayasemi) [reply settings/templates/panels/admin/filesharing.php]: Yes, the first one is forbidden by php-cs-fixer and when I used multi-line `if`, codecov complained about low coverage in the template... I will try your second option. Let's see what will codecov say? By the way, the whole file mess in terms of readability.

**Comment by jvillafanez** [comment]
Screen Shot 2019-04-04 at 12 50 54

Not sure if it's intended, but shouldn't the "enforce expiration date" checkbox be below the date?

**Comment by karakayasemi** [comment]
> Screen Shot 2019-04-04 at 12 50 54
> 
> Not sure if it's intended, but shouldn't the "enforce expiration date" checkbox be below the date?

I did not make any change about it. The issue and PR are only about sub-entry location.  We can ask it to @pmaier1 and @PVince81 .

**Comment by PVince81** [comment]
@karakayasemi yeah, you could just fix the wrapping to have it go to the next line

**Comment by pmaier1** [comment]
Yes, agreed, move it below, please.

**Comment by karakayasemi** [comment]
"enforce expiration date" checkbox carried to newline by adding `<br/>` after first input.

**Review by jvillafanez — APPROVED** [review]

**Comment by PVince81** [comment]
@karakayasemi please backport

**Pull request merged**